x-pack/plugins/lens/public/xy_visualization/xy_config_panel/color_picker.tsx

```diff
@@ -60,6 +64,10 @@ export const ColorPicker = ({
     const datasource = frame.datasourceLayers[layer.layerId];
     const sortedAccessors: string[] = getSortedAccessors(datasource, layer);
 
+    if (layer.layerType === layerTypes.THRESHOLD) {
```

Comment:
nit: could we move this `if` to line 63 to not calculate datasource or sortedAccessors unnecessarily?